### PR TITLE
Add asyncauth tests

### PR DESCRIPTION
**Pull request by mhzajac**
https://jaroop.atlassian.net/browse/JCOR-219
https://jaroop.atlassian.net/browse/JCOR-220

### CODE REVIEW

**Comment by mhzajac** [comment]
The only issue I'm having is that for whatever reason tests fail when running them a second time locally with really weird mockito errors. Running them when a freshly started sbt always works.

**Comment by mhzajac** [comment]
Fixed.

File: build.sbt [hunk 1]
```diff
@@ -44,10 +43,14 @@ def scalacOptionsVersion(scalaVersion: String) = {
...
   .settings(
-    baseSettings,
-    libraryDependencies += "com.typesafe.play" %% "play" % playVersion % "provided",
-    libraryDependencies += "com.typesafe.play" %% "play-cache" % playVersion % "provided",
     name := appName,
+    baseSettings,
+    libraryDependencies ++= Seq(
+      "com.typesafe.play" %% "play" % playVersion % "provided",
+      "com.typesafe.play" %% "play-cache" % playVersion % "provided",
+      "org.specs2" %% "specs2-core" % "3.9.4" % "test",
```
Comment (btmcnellis) [inline build.sbt]: This PR has been sitting around for long enough that there are now two newer versions, 3.9.5 and 4.0.0. The latter has some breaking changes around `await` that I think cause some of the tests to fail.

**Comment by mhzajac** [comment]
Updated specs2 to 3.9.5.

**Pull request merged**